python/tank/platform/validation.py

```diff
@@ -379,16 +391,39 @@ def validate(self, settings):
         # first sanity check that the schema is correct
         validate_schema(self._display_name, self._schema)
         
-        # Ensure that all required keys are in the settings and that the
-        # values are appropriate.
+        # Ensure that all keys are in the settings or have a default value in
+        # the manifest. Also make sure values are appropriate.
         for settings_key in self._schema:
             value_schema = self._schema.get(settings_key, {})
-            
-            # make sure the required key exists in the environment settings
-            if settings_key not in settings:
-                raise TankError("Missing required key '%s' in settings!" % settings_key)
-            
-            self.__validate_settings_value(settings_key, value_schema, settings[settings_key])
+
+            if settings_key in settings:
+                # value exists in the settings. use it.
+                settings_value = settings[settings_key]
+            else:
+                # Use the fallback default with an unlikely to be used value to
+                # detect cases where there is no default value in the schema.
+                try:
+                    settings_value = resolve_default_value(value_schema,
+                        raise_if_missing=True)
+                except TankError:
```

Comment:
wondering if this should be a special exception rather than TankError? Right now, we assume that this TankError means that the method could not identify a default value, but I am guessing there could be other reasons a tank error may be raise?